music/src/core/melodies.ts

```diff
@@ -254,3 +254,65 @@ export class MelodyShape implements MelodyControl {
     return buffer.toTensor().as2D(numSteps, 3);
   }
 }
+
+/**
+ * Register control signal. Extracts a tensor with one-hot register indicator,
+ * constant across time steps.
+ */
+export class MelodyRegister implements MelodyControl {
+  readonly boundaryPitches: number[];
+  readonly depth: number;
+
+  constructor(boundaryPitches: number[]) {
+    this.boundaryPitches = boundaryPitches;
+    this.depth = boundaryPitches.length + 1;
+  }
+
+  private meanMelodyPitch(melody: Melody): number {
+    // Compute mean melody pitch, weighted by time steps.
```

Comment:
weighted by non-silent time steps

Reply:
Edited to clarify; I think "non-silent" is implied by "pitch" but I made it clearer that it's explicitly weighting each pitch by how long it is sustained.